src/embed/init.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable no-underscore-dangle, camelcase */
+import { DataLayerObserver } from '../observer';
+
+/*
+This is where we initialize the DataLayerObserver from this info:
+
+// If true, output from rules are sent only to `console`
+// Default is false
+window['_dlo_preview'] = false;
+
+// If true, rules will run on page load in addition to when there are changes
+// Default is true
+window['_dlo_readOnLoad'] = false;
+
+// Anything on `window` that starts with `_dlo_rules` is read as a rules array
+window['_dlo_rules'] = [
+// rules can go here
+];
+
+window['_dlo_rulesFromAdTeam'] = [
+// rules can also go here
+];
+
+window['_dlo_rulesFromOpsTeam'] = [
+// rules can also go here
+];
+
+*/
+
+function _dlo_collectRules(): any[] {
+  const results: any[] = [];
+  Object.getOwnPropertyNames(window).forEach((propName) => {
+    if (propName.startsWith('_dlo_rules') === false) return;
+
+    const prop = (window as { [key: string]: any })[propName];
+    if (Array.isArray(prop) === false) {
+      /* eslint-disable no-console */
+      console.warn(`window[${propName}] is not an array so will be ignored`);
```

Comment:
Can we use `Logger.getInstance().warn('No rules for the Data Layer Observer')`?  That will by default use the `ConsoleAppender`, but I think what we also want to do is set up FullStory as the log appender.  The `logger.spec.ts` has a unit test to add FullStory:

```
const logger = Logger.getInstance();
    logger.appender = {
      log(event: LogEvent) {
        // eslint-disable-next-line camelcase
        const { level: level_int, message: message_str, datalayer: datalayer_str } = event;
        FS.event('Data Layer Observer', { level_int, message_str, datalayer_str }, 'dataLayerObserver');
      },
    };
```

Can we do something similar by creating a `_dlo_log` property or similar to assign a custom function.  Our default snippet would then provide the correct implementation to use FullStory.